### PR TITLE
Update simgrid 3.19.1 -> 3.20 + parallel tests + enable darwin

### PR DESCRIPTION
**Pull request by mickours**
###### Motivation for this change

Enable the build on Darwin.
Speed up testing

###### Things done

I use this update to remove ``elfutils`` from the ``nativeBuildInputs`` because it was only needed for Model Checking. This should enables the build on Darwin without this option which is activated by default.

Also I add a falg to Ctest command to enable parallel run of the tests  

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/misc/simgrid/default.nix [hunk 1]
```diff
@@ -83,13 +83,21 @@ stdenv.mkDerivation rec {
...
-    ctest --output-on-failure -E smpi-replay-multiple
+
+    if [ "$enableParallelBuilding" = 1 ]; then
+      if [ ''${NIX_BUILD_CORES:-0} = 0 ]; then
+        NB_CORES="$(nproc)"
+      else
+        NB_CORES="$NIX_BUILD_CORES"
+      fi
+    fi
+    ctest -j $NB_CORES --output-on-failure -E smpi-replay-multiple
```
Comment (LnL7) [inline pkgs/applications/science/misc/simgrid/default.nix]: Doesn't -j0 work here, like with make?

**Comment by LnL7** [comment]
@GrahamcOfBorg build simgrid

File: pkgs/applications/science/misc/simgrid/default.nix (hunk 1 above)
Comment (mickours) [reply pkgs/applications/science/misc/simgrid/default.nix]: I'm not sure, nothing is specified for "0" value in the ctest documentation (https://cmake.org/cmake/help/v3.0/manual/ctest.1.html), so I did it this way.

**Review by matthewbauer — APPROVED** [review]

**Review by matthewbauer — CHANGES_REQUESTED** [review]
Avoid setting custom NB_CORES values. NIX_BUILD_CORES should just work everywhere (and if it doesn't we should handle that in CMake not here).

**Comment by matthewbauer** [comment]
Yeah I think the ctest stuff is good just that it shouldn't need the if statement. Just like this is what we do in other places:

```
     ctest -j $NIX_BUILD_CORES --output-on-failure -E smpi-replay-multiple
```

**Comment by matthewbauer** [comment]
The problem is that nproc isn't really a good guess. If you have 8 cores, but 4 jobs running on them, you don't want to run the tests at ```-j 8```. Anyway we handle this logic on setup.sh:

https://github.com/NixOS/nixpkgs/blob/e46e7f44c22fcf328d62c10152926852e6e4b896/pkgs/stdenv/generic/setup.sh#L624-L631

**Comment by mickours** [comment]
Thanks for the explanation, I've simplify the ctest command as you suggests.

And sorry for being so insistent, I'm sometimes a bit stubborn when I don't understand ;)

**Comment by mpickering** [comment]
Seems like your concerns were dealt with @matthewbauer ?

**Review by matthewbauer — APPROVED** [review]

**Comment by matthewbauer** [comment]
@GrahamcOfBorg build simgrid

**Pull request merged**